state/mysql/mysql.go

```diff
@@ -305,6 +371,44 @@ func (m *MySQL) ensureStateTable(ctx context.Context, stateTableName string) err
 		}
 	}
 
+	// Check if expiredate column exists - to cater cases when table was created before v1.11.
+	columnExists, err := columnExists(ctx, m.db, schemaName, stateTableName, "expiredate", m.timeout)
+	if err != nil {
+		return err
+	}
+
+	if !columnExists {
+		m.logger.Infof("Adding expiredate column to MySql state table '%s'", stateTableName)
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(
+			`ALTER TABLE %s ADD COLUMN expiredate TIMESTAMP NULL;`, stateTableName))
+		if err != nil {
+			return err
+		}
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(
+			`CREATE INDEX expiredate_idx ON %s (expiredate);`, stateTableName))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *MySQL) ensureMetadataTable(ctx context.Context, schemaName, metaTableName string) error {
+	exists, err := tableExists(ctx, m.db, schemaName, metaTableName, m.timeout)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		m.logger.Info("Creating MySQL metadata table")
+		_, err = m.db.ExecContext(ctx, fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
```

Comment:
Nit: I would prefer name `key` rather than `id` for the primary key, given this is a "key-value".

Reply:
Name `key` is not allowed. I started off with `key` only but it gives an error.